apiserver/Volumes.md

```diff
@@ -0,0 +1,147 @@
+specific size and access modes (e.g., they can be mounted `ReadWriteOnce`, `ReadOnlyMany` or `ReadWriteMany`).
+
+The caveat of using PVC volumes is that the same PVC is mounted to all nodes. As a result only PVCs with access
+mode `ReadOnlyMany` can be used in this case.
+
+The code below gives an example of PVC volume definition:
+
+````
+{
+	"name": "pvc",              # unique name
+	"mountPath": "/tmp/pvc",    # mounting path
+	"volumeType": 0,            # volume type - PVC
+	"mountPropagationMode": 2,  # mount propagation mode - bidirectional
+	"readOnly": false           # read only
+}
+````
+
+## Ephemeral volumes
+
+Some application need additional storage but don't care whether that data is stored persistently across restarts. For 
+example, caching services are often limited by memory size and can move infrequently used data into storage that is 
+slower than memory with little impact on overall performance. Ephemeral volumes are designed for these use cases. 
+Because volumes follow the Pod's lifetime and get created and deleted along with the Pod, Pods can be stopped and 
+restarted without being limited to where some persistent volume is available.
+
+Although there are several option of ephemeral volumes, here we are using generic ephemeral volumes, which can be 
+provided by all storage drivers that also support persistent volumes. Generic ephemeral volumes are similar to emptyDir 
+volumes in the sense that they provide a per-pod directory for scratch data that is usually empty after provisioning. 
+But they may also have additional features:
+
+* Storage can be local or network-attached.
+* Volumes can have a fixed size that Pods are not able to exceed.
+
+The code below gives an example of ephemeral volume definition:
+
+````
+{
+	"name": "ephemeral",            # unique name
+	"mountPath": "/tmp/ephemeral"   # mounting path,
+	"mountPropagationMode": 0,      # mount propagation mode - None
+	"volumeType": 2,                # volume type - ephemeral
+	"storage": "5Gi",               # disk size
+	"storageClass": "default"       # storage class - optional
+	AccessMode: 0                   # access mode RWO - optional
+}
+````
+
+## Config map volumes
+
+A ConfigMap provides a way to inject configuration data into pods. The data stored in a ConfigMap can be referenced in 
+a volume of type configMap and then consumed by containerized applications running in a pod.
+
+When referencing a ConfigMap, you provide the name of the ConfigMap in the volume. You can customize the path to use 
+for a specific entry in the ConfigMap.
+
+The code below gives an example of config map volume definition:
+
+````
+{
+    "name":"code-sample",               # Unique name
+    "mountPath":"/home/ray/samples",    # mounting path
+    "volumeType":3,                     # volume type - config map
+    "source":"ray-job-code-sample",     # config map name
```

Comment:
I feel like we should mention the configMap can be created first using `kubectl -f test/job/code.yaml` some where.
```suggestion
    "source":"ray-job-code-sample",     # config map name. kubectl -f test/job/code.yaml
```

Reply:
Not really. This is just an example. It assumes, that the configMap exists and it is referenced here. It says above:
````
The data stored in a ConfigMap can be referenced in a volume
````